### PR TITLE
Pricefeed price simulation refactor

### PR DESCRIPTION
**Pull request by nddeluca**
Refactors the pricefeed simulation to use PriceGenerator object that deterministically randomizes price movements.

### CODE REVIEW

**Comment by nddeluca** [comment]
Thanks Denali! The goal here is to just make it deterministic, agree we need to improve the problem cases and add in more adversarial cases to the price changes. Added a card to the hopper. Going to mark this PR as ready for review

File: x/pricefeed/simulation/operations.go [hunk 1]
```diff
@@ -198,3 +129,93 @@ func getExpiryTime(ctx sdk.Context) (t time.Time) {
 	// need to use the blocktime from the context as the context generates random start time when running simulations
 	return ctx.BlockTime().Add(time.Second * 1000000)
 }
+
+// PriceGenerator allows deterministic price generation in simulations
+type PriceGenerator struct {
```
Comment (fedekunze) [inline x/pricefeed/simulation/operations.go:134]: I'd move this struct and its functions to a new file `x/pricefeed/simulation/types.go` to keep logic clean from type definitions 👍

Comment (karzak) [reply x/pricefeed/simulation/operations.go:134]: 👍  to moving the type out

Comment (nddeluca) [reply x/pricefeed/simulation/operations.go]: Great idea 💯

**Pull request merged**